code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -589,3 +589,28 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			user << "<span class='warning'>You need to dry this first!</span>"
 	else
 		..()
+///////////////////
+//Strike anywhere//
+///////////////////
+/obj/item/clothing/mask/cigarette/strikeanywhere
+	name = "strike anywhere cigarette"
+	desc = "A cigarette infused with a proprietary compound that ignites when struck. Smells of magnesium."
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/afterattack(atom/target, mob/user, proximity)
+	if(!lit && smoketime > 0)
+		user.visible_message("[user] strikes [target] with the [src] and it ignites!", "<span class='notice'>You strike [target] with the [src] and it ignites!</span>")
+		light()
+	..()
+
+/obj/item/clothing/mask/cigarette/strikeanywhere/throw_impact(atom/hit_atom)
+	if(!lit && smoketime > 0)
+		hit_atom.visible_message("The [src] strikes [hit_atom] and ignites!")
+		light()
+	..()

```

Comment:
If you are an admin on your test server you can use aghost and then click drag your ghost onto a mob to take control of that mob. Another way is to use rightclick on a mob to check its variables (view variables window) and use the "Assume Direct Control" option in the drop down menu on the top right of the window.